vtkext/private/module/vtkF3DImguiConsole.cxx

```diff
@@ -21,6 +24,151 @@ struct vtkF3DImguiConsole::Internals
   std::array<char, 256> CurrentInput = {};
   bool NewError = false;
   bool NewWarning = false;
+  std::pair<size_t, size_t> Completions{ 0,
+    0 }; // Index for start and length of completions in Logs
+  std::function<std::vector<std::string>(const std::string& pattern)>
+    GetCommandsMatchCallback; // Callback to get the list of commands matching pattern
+  std::vector<std::string> CommandHistory;
+  std::pair<std::string, int> LastInput; // Last input before navigating history
+  int CommandHistoryIndex = -1;          // Current index in command history navigation
+  /**
+   * Clear completions from the logs
+   */
+  void ClearCompletions()
+  {
+    if (Completions.second > 0)
+    {
+      Logs.erase(Logs.begin() + Completions.first, Logs.begin() + Completions.second);
+      Completions.second = 0;
+    }
```

Comment:
We do add `this->` for class members for clarity.  
Here's a suggestion for this code block, but there are other locations below.

```suggestion
    if (this->Completions.second > 0)
    {
      this->Logs.erase(this->Logs.begin() + this->Completions.first, this->Logs.begin() + this->Completions.second);
      this->Completions.second = 0;
    }
```